Add option to restrict the desired encoding when searching for episodes.

sammys opened the pull request:
I'm creating a pull to notify everyone that this is in progress. I welcome people to test this out (backup your database first) and also to suggest any improvements that can be made to this particular feature.

Due to the size of the patch I will limit the scope to restricting searches **only**. If you have other encodings that need to be supported along with file extensions then please notify me.

I have not tested mass updates nor saving the encoding when episodes are downloaded or postprocessed. That needs a bit more testing than I have time for right now. Please assist if you can.

sammys committed:
Add option to restrict the desired encoding when searching for episodes.

thezoggy commented:
'encoding' is the wrong term for this. 'codec' is the correct one. and 'xvid' is no more as of the end of 2012. so i see little reason to add this.. it would only cause issues for people trying to backfill. there was only a brief period where you saw both codecs being used..

sammys committed:
Small fixes for encoding restriction

sammys committed:
Use correct nomenclature: codec.
Remove remnants of the previous implementation.
Fix a few errors.

sammys commented:
Hi @thezoggy. Thanks for taking a look at this pull request. As you said, codecs only causes issues for people trying to backfill. Use case:

> Person 'A' solely uses iTunes for one's TV show playback so, for all shows, 'A' wants to automatically download x264 releases only. Once they are downloaded 'A' will switch the restriction to Any and have their postprocessor transcode them.

It's particularly handy for shows like Saturday Night Live, which have many seasons. Let's say that the person would like to only download x264 releases and ignore XviD. Another use case:

> A show has already been downloaded in XviD and transcoded. The codec can then be restricted to x264 and switched to Wanted. Any subsequent blu-ray releases will then be downloaded and overwritten.

Not sure if the overwriting part is implemented yet, hehe.

It may not be useful to you but there are people with these unique use cases. The code is not very complex and it won't add much maintenance overhead.

To meet your requirements I have modified all code to use the term codec instead of encoding.

Thanks again for considering my contribution to Sick Beard. Nice work so far btw.

sammys committed:
Determine the codec from the cached nzb name rather than store it in the DB.

sammys committed:
Fix problems when adding a new show.

sammys committed:
Merge pull request #1 from midgetspy/development

Merge in upstream development changes.

sammys committed:
Merge branch 'development' into restrict-encoding

Conflicts:
	sickbeard/providers/generic.py
	sickbeard/tvcache.py

sammys committed:
Fix use of wrong dict for zero codec value when displaying show.
Cast codec value to int before saving a TV show.

sammys committed:
Small fixes for codec restriction.

sammys committed:
Change encoding to codec in addShowOptions.js

sammys committed:
Fix incorrect parameter order for addShow queue action and change "encoding" to "codec".

sammys committed:
Make codec parameter optional in TVShow.wantEpisode().

sammys committed:
Final fix for codec restrictions: Display of freshly added show has unknown codec displayed even though DB has correct value.
